src/test/java/io/jenkins/plugins/analysis/core/graphs/SeriesBuilderTest.java

```diff
@@ -0,0 +1,317 @@
+package io.jenkins.plugins.analysis.core.graphs;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.assertj.core.util.Lists;
+import org.jfree.data.category.CategoryDataset;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static io.jenkins.plugins.analysis.core.graphs.assertj.Assertions.*;
+import io.jenkins.plugins.analysis.core.quality.AnalysisBuild;
+import io.jenkins.plugins.analysis.core.quality.StaticAnalysisRun;
+import static java.util.Arrays.*;
+import static org.mockito.Mockito.*;
+
+
+class SeriesBuilderTest {
+
+
+    private static final GraphConfiguration CONFIG_BUILD_COUNT_NONE = createWithBuildCount(0);
+    private static final GraphConfiguration CONFIG_BUILD_COUNT_ONE = createWithBuildCount(1);
+    private static final GraphConfiguration CONFIG_BUILD_COUNT_TWO = createWithBuildCount(2);
+    private static final GraphConfiguration CONFIG_BUILD_DATE = createWithBuildDate();
+
+    private static final DateTime DAY = new DateTime(0, DateTimeZone.UTC).plusDays(1);
+    private static final DateTime PREVIOUS_DAY = DAY.minusDays(1);
+    private static final DateTime SAME_DAY = DAY.plusHours(4);
+    private static final DateTime NEXT_DAY = DAY.plusDays(1);
+
+    private static final StaticAnalysisRun RUN_PREVIOUS_DAY = createRun(1, PREVIOUS_DAY);
+    private static final StaticAnalysisRun RUN_DAY = createRun(2, DAY);
+    private static final StaticAnalysisRun RUN_SAME_DAY = createRun(3, SAME_DAY);
+    private static final StaticAnalysisRun RUN_NEXT_DAY = createRun(4, NEXT_DAY);
+
+    private static final List<Integer> FIRST_SERIES = series(0, 1, 2);
+    private static final List<Integer> SECOND_SERIES = series(3, 4, 5);
+    private static final List<Integer> FORTH_SERIES = series(9, 10, 11);
```

Comment:
Fehlt die 3?

Reply:
Bewusst nicht. Der vierte Run wird einmal benötigt, da in Test "build date, never too old, average same days" der Run hinter dem Average-Fall überprüft wird. Ich dachte, 0, 1 und 2 sollte genügen, da sich ein Mehrwert hinter 3 Runs für mich nicht erschloss. Kann ich auch gerne nachbessen, falls Sie anderer Meinung sind.

Reply:
Für die Testabdeckung ist das prima so, nur wenn man die Variablen Namen liest, wundert man sich warum es kein THIRD aber ein FORTH gibt.

Reply:
Nun habe ich auch Ihre Anmerkung verstanden, wie Sie Sie meinten ;-) Ist bereits ausgebessert.